Controllers/BranchesController.cs

```diff
@@ -0,0 +1,215 @@
+            }
+
+            return View(branchesVM);
+        }
+
+        // GET: Entities
+        public async Task<IActionResult> ListBranches(int? id)
+        {
+            var applicationDbContext = _context.Branches
+                .Include(b => b.Entity)
+                .Include(b => b.Contact)
+                .Where(m => m.EntityId == id);
+
+            var branches = await applicationDbContext.ToListAsync();
+            var branchesVM = new List<BranchFormViewModel>();
+
+            foreach (var branch in branches)
+            {
+                var branchVM = new BranchFormViewModel();
+
+                branchVM = _mapper.Map<BranchFormViewModel>(branch);
+
+                branchesVM.Add(branchVM);
+            }
+
+            return View(branchesVM);
+        }
+
+        // GET: Branches/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var branch = await _context.Branches
+                .Include(b => b.Contact)
+                .Include(b => b.Entity)
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (branch == null)
+            {
+                return NotFound();
+            }
+
+            var branchVM = _mapper.Map<BranchFormViewModel>(branch);
+
+            return View(branchVM);
+        }
+
+        // GET: Branches/Create
+        public IActionResult Create()
+        {
+            ViewData[nameof(Branch.EntityId)] = new SelectList(_context.Entities, nameof(Entity.Id), nameof(Entity.Name));
+            return View();
+        }
+
+        // POST: Branches/Create
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create(BranchFormViewModel branchVM)
+        {
+            if (ModelState.IsValid)
+            {
+                var branch = _mapper.Map<Branch>(branchVM);
+
+                _context.Add(branch);
+                await _context.SaveChangesAsync();
+                return RedirectToAction(nameof(Index));
+            }
+            ViewData[nameof(Branch.EntityId)] = new SelectList(_context.Entities, nameof(Entity.Id), nameof(Entity.Name));
+
+            return View(branchVM);
+        }
+
+        // GET: Branches/Edit/5
+        public async Task<IActionResult> Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var branch = await _context.Branches
+                .Include(b => b.Contact)
+                .Include(b => b.Entity)
+                .FirstOrDefaultAsync(m => m.Id == id);
+
+            //var branch = await _context.Branches.FindAsync(id);
+
+            if (branch == null)
+            {
+                return NotFound();
+            }
+            ViewData[nameof(Branch.EntityId)] = new SelectList(_context.Entities, nameof(Entity.Id), nameof(Entity.Name));
```

Comment:
This I guess need to be still refactored.
Please watch 2:57 of
https://www.udemy.com/course/the-complete-aspnet-mvc-5-course/learn/lecture/4853586#overview 
I suggest you to add CreateBranchViewModel type, that contains inside SelectList, that can be consumed by Razor.

Reply:
Or maybe better than constructing SelectList inside CreateBranchViewModel is just do as he did there https://www.udemy.com/course/the-complete-aspnet-mvc-5-course/learn/lecture/4924164#overview - just add list of object and then use @Html.DropDownListFor helper method.